numpy/random/src/distributions/distributions.c

```diff
@@ -1,9 +1,27 @@
-#include "numpy/random/distributions.h"
+
+/*
+ * RAND_INT_TYPE is used to share functions between _generator.pyx and
+ * legacy mtrand.pyx (where it is expected to be "long").
+ *
+ * If you change a function in this file, then the original unmodified
+ * copy must be retained for use in RandomState via mtrand.pyx by copying to
+ * the legacy-distributions.c source file.
+ */
+
+#ifndef RAND_INT_TYPE
+  #define RAND_INT_TYPE int64_t
+  #define RAND_INT_MAX INT64_MAX
+  #include "numpy/random/distributions.h"
+#else
+  /* For internal numpy legacy use only */
+  #include "include/distributions-long.h"
```

Comment:
Is "long" a good name? Maybe legacy, or legacy-internal?

Reply:
fixing to `legacy`

Reply:
redoing this: the use of that `#define` is too confusing. I am separating it into two files: `src/distributions/distributions.c` and 'src/legacy/legacy-distributions.c`

Reply:
I reverted this change.